COMMIT_GUIDELINES.md

```diff
@@ -0,0 +1,141 @@
+# General philosophy
+
+- **Each change goes into its own commit**. If you want to summarize what you did with this commit (in the commit message)
+and you start using the word "and", you probably want to split the commit up into 2 or more individual commits.
+
+- **Each commit should compile** (try to make each commit self-contained so that it is possible to compile the code at
+each commit)
+
+- **Take your time** when composing the commit message. In order to have a good git-history the commit messages are essential.
+Also if you put effort into the commit-message, you'll save work when creating a PR as the description is already available.
+
+
+# Commit message
+
+Commit messages must follow the scheme
+
+```
+TYPE(Scope): Summary
+
+Message Body
+
+Footer
+```
+
+The blank lines in between are mandatory. A commit **must** include a `TYPE` and a `Summary` and **may**
+additionally contain any of the other components listed here.
+
+
+## Subject line
+
+The first line ("Subject line") should not exceed 50-70 characters. This is what gets displayed on GitHub at first glance, so it
+should contain the most important information. In order for it to be as short and precise as possible, there is the `TYPE` and
+optionally a `Scope`. With these it should already be clear what this commit is about in general. The short summary should
```

Comment:
Ah, I really should have explained myself better.

What I like about this overall explanation is that you're highlighting the parts of the commit message with code formatting.

For example, you are doing this:

```
there is the `TYPE` and optionally a `Scope`.
```

Instead of this:
```
there is the TYPE and optionally a Scope.
```

Which I think connects really nicely with the above diagram that shows the structure of the ideal commit message.

So the change that I meant to ask for was to make

```
The short summary should
```

be written like this:

```
The short `Summary` should
```

@Krzmbrzl :3